src/main/java/com/enterprise/fastfoodapplication/FastFoodController.java

```diff
@@ -112,19 +112,18 @@ public ResponseEntity fetchFoodById(@PathVariable("id") int id){
      * @return the newly created food object.
      */
     @PostMapping(value="/Food", consumes ="application/json", produces = "application/json")
-    public ResponseEntity createFood(@RequestBody Food food){
-        try {
-            Food createdFood = foodService.createFoodItem(food);
-            HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.APPLICATION_JSON);
-            return new ResponseEntity(createdFood, headers, HttpStatus.OK);
-        } catch (Exception e) {
-            e.printStackTrace();
-            logger.log(Level.WARNING, "Failed to create food item");
-            return  new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
-            /*TO DO Logging*/
-        }
+    public Food createFood(@RequestBody Food food){
+    Food newFood = null;
+    try {
+        foodService.createFoodItem(food);
+    } catch (Exception e){
+        logger.log(Level.WARNING, "Failed to create item of" 
+                + " ID: " + food.getFoodId()
+                + " for food " + food.getFoodName());
     }
+    return food;
```

Comment:
Careful here.  food is initialized to null, so what happens if an exception is thrown?
The exception is logged, and discarded, and then a null is returned.
That puts you in a worse off position, because you'll end up with an ambiguous NullPointerException later.
This is a case where it might be better to just allow the exception to be thrown, so that the calling method knows what happened.